server/utils/azureAppinsights.test.ts

```diff
@@ -0,0 +1,82 @@
+import { DataTelemetry, EnvelopeTelemetry } from 'applicationinsights/out/Declarations/Contracts'
+import { addUserDataToRequests, ContextObject } from './azureAppInsights'
+
+const user = {
+  activeCaseLoadId: 'LII',
+  username: 'test-user',
+}
+
+const createEnvelope = (properties: Record<string, string | boolean>, baseType = 'RequestData') =>
+  ({
+    data: {
+      baseType,
+      baseData: { properties },
+    } as DataTelemetry,
+  } as EnvelopeTelemetry)
+
+const createContext = (username: string, activeCaseLoadId: string) =>
+  ({
+    'http.ServerRequest': {
+      res: {
+        locals: {
+          user: {
+            username,
+            activeCaseLoadId,
+          },
+        },
+      },
+    },
+  } as ContextObject)
+
+const context = createContext(user.username, user.activeCaseLoadId)
+
+describe('azure-appinsights', () => {
+  describe('addUserDataToRequests', () => {
+    it('adds user data to properties when present', () => {
+      const envelope = createEnvelope({ other: 'things' })
+
+      addUserDataToRequests(envelope, context)
+
+      expect(envelope.data.baseData.properties).toStrictEqual({
+        ...user,
+        other: 'things',
+      })
+    })
+
+    it('handles absent user data', () => {
+      const envelope = createEnvelope({ other: 'things' })
+
+      addUserDataToRequests(envelope, createContext(undefined, user.activeCaseLoadId))
+
+      expect(envelope.data.baseData.properties).toStrictEqual({ other: 'things' })
+    })
+
+    it('returns true when not RequestData type', () => {
+      const envelope = createEnvelope({}, 'NOT_REQUEST_DATA')
+
+      const response = addUserDataToRequests(envelope, context)
+
+      expect(response).toStrictEqual(true)
```

Comment:
should this be in every test maybe?

Reply:
I just copied from the other PR and had a look at the tests, as you said I don't think it really matters :)